apps/meteor/app/channel-settings/server/methods/saveRoomSettings.js

```diff
@@ -241,8 +241,8 @@ const settingSavers = {
 	encrypted({ value, room, rid, user }) {
 		saveRoomEncrypted(rid, value, user, Boolean(room.encrypted) !== Boolean(value));
 	},
-	favorite({ value, rid }) {
-		Rooms.saveFavoriteById(rid, value.favorite, value.defaultValue);
+	favorite({ value, rid, user }) {
+		Subscriptions.updateOne({ rid, 'u._id': user._id }, { $set: { f: value ?? true } });
```

Comment:
value is an object with two properties (`favorite` and `defaultValue`)

Reply:
Updated